9627 stin docket record index

jamesobrooks opened the pull request:
# [9627](https://github.com/flexion/ef-cms/issues/9627)

When adding the STIN on the wizard initially, we set the index to 0. When adding initial documents we were generating the next index number when we should not assign a real index number.

jamesobrooks committed:
9627: Set index to 0 if eventCode is STIN

When adding the STIN on the wizard initially,
we set the index to 0. When adding initial documents we were
generating the next index number when we should not assign
a real index number.

jamesobrooks committed:
9627: Do not include STIN on docket record

isOnDocketRecord should always be false for STIN. This is
reflected in the new error we throw if we get into addDocketEntry
with it set to true. We still need to set the index to 0.

jamesobrooks committed:
9627: Properly test exceptional behavior

jamesobrooks commented on shared/src/business/entities/cases/Case.addDocketEntry.test.js:
```diff
@@ -4,7 +4,25 @@ const {
 
 describe('addDocketEntry', () => {
-  it('attaches the docket entry to the case', () => {
+  it('should throw when docket entry is a STIN', () => {
```
I'm now wondering if these titles should be different.
```suggestion
  it('should throw when docket entry to be added to the docket record is a STIN', () => {
```

mrinsin replied:
I feel like that may not be necessary as the method under test is conveying the adding of the docket entry! (but thats just my opinion)

jamesobrooks commented on shared/src/business/entities/cases/Case.addDocketEntry.test.js:
```diff
@@ -24,4 +42,29 @@ describe('addDocketEntry', () => {
     });
   });
+
+  it("should assign docket entry index of '0' to the STIN", () => {
```
Following with the clarification of the above test:
```suggestion
  it("should assign docket entry index of '0' to the STIN when not being added to the docket record", () => {
```

mrinsin replied:
This rename makes sense to me!!

jamesobrooks committed:
Merge branch 'staging' into 9627-stin-docket-record-index

jamesobrooks committed:
Merge branch 'staging' into 9627-stin-docket-record-index

mrinsin commented on shared/src/business/useCaseHelper/initialFilingDocuments/updateInitialFilingDocuments.js:
```diff
@@ -92,9 +92,10 @@ exports.updateInitialFilingDocuments = async ({
   caseEntity,
   caseToUpdate,
 }) => {
+  const PETITION_KEY = 'petitionFile';
```
Nice, I wonder if this string is used elsewhere in the code and if this would be worth moving to `EntityConstants`

jamesobrooks replied:
@mrinsin there is a use in `CaseInternal.js` for validation, but there are several similar strings in use there too. Maybe a separate PR for that change if we want it? I don't feel strongly either way.

mrinsin replied:
I don't feel too strongly about the changes being in this PR! 👍 to different PR

jamesobrooks committed:
Test name clarification

Make the test name clear that the STIN will not be on the docket record.

jamesobrooks committed:
Test name clarification

Make it clear that if trying to add the STIN to the docket record is exceptional behavior.

mmarcotte committed:
Merge branch 'staging' into 9627-stin-docket-record-index
